satpy/node.py

```diff
@@ -463,7 +455,8 @@ def _find_dependencies(self, dataset_key, **dfilter):
 
         # 0 check if the *exact* dataset is already loaded
         try:
-            node = self.getitem(dataset_key)
+            dsid = create_filtered_dsid(dataset_key, **dfilter)
+            node = self.getitem(dsid)
```

Comment:
I'm not sure about this. Checking if the *exact* dataset is meant to see if an exact `DatasetID` exists. I don't think it should be modified by `dfilter` parameters.

Reply:
Well, it's the point of my resolution PRs that the requested dataset with the correct resolution should be returned. Or is that not the point of the `resolution` keyword in `Scene.load` ?

Reply:
I mean that passing `resolution` and a name should be equivalent to passing a full fledged `DatasetID` right ?

Reply:
Yes, but if something gets to `_find_dependencies` and passes a `DatasetID` that means that's the *exact* ID that should be used. However, I can see that if that DatasetID (in composite loading, etc) hasn't been updated with the `dfilter` then it would likely return the wrong resolution.

The problem, from what I can tell, with the way it is implemented now is that if an exact DatasetID is provided to `_find_dependencies` it's resolution (and other parameters) are overwritten with `dfilter` even if they were specified exactly by the composite config. Another case would be a user specifying `scn.load([DatasetID(name='X', resolution=500), 'Y', 'Z'], resolution=1000)`, what should happen then?

Reply:
ok, I could fix it so that the create_filtered_dsid gives the priority to the existing DatasetID instance if it is filled with other things than None. Would that work ?

Reply:
I *think* so